api/Updated-Metrics.md

```diff
@@ -0,0 +1,326 @@
+
+### Existing Syntax
+
+Our current API does not support evaluating metrics on intersections of sensitive features (e.g. "black and female", "black and male", "white and female", "white and male").
+To achieve this, users currently need to write something along the lines of:
+```python
+>>> A_combined = A['SF 1'] + '-' + A['SF 2']
+
+>>> accuracy_score_group_summary(y_true, y_pred, sensitive_features=A_combined)
+{ 'overall': 0.4, by_groups : { 'B-M':0.4, 'B-N':0.5, 'B-P':0.5, 'C-M':0.5, 'C-N': 0.6, 'C-P':0.7 } }
+```
+This is unecessarily cumbersome.
+It is also possible that some combinations might not appear in the data (especially as more sensitive features are combined), but identifying which ones were not represented in the dataset would be tedious.
+
+
+### Proposed Change
+
+If `sensitive_features=` is a DataFrame (or list of Series.... exact supported types are TBD), we can generate our results in terms of a MultiIndex. Using the `A` DataFrame defined above, a user might write:
+```python
+>>> result = group_summary(skm.accuracy_score, y_true, y_pred, sensitive_features=A)
+>>> result.by_groups
+           accuracy_score
+SF 1 SF 2
+B    M               0.50
+     N               0.40
+     P               0.55
+C    M               0.45
+     N               0.70
+     P               0.63
+```
+If a particular combination of sensitive features had no representatives, then we would return `None` for that entry in the Series.
+Although this example has passed a DataFrame in for `sensitive_features=` we should aim to support lists of Series and `numpy.ndarray` as well.
+
+The `differences()` and `ratios()` methods would act on this Series as before.
+
+## Conditional (or Segmented) Metrics
+
+For our purposes, Conditional Metrics (alternatively known as Segmented Metrics) are specified separately from the sensitive features, since for users, they add columns to the result.
+Mathematically, they behave like additional sensitive features.
+
+### Existing Syntax
+
+Not supported.
+Users would have to devise the required code themselves
+
+### Proposed Change
+
+The `GroupedMetric` constructor will need an additional argument `conditional_features=` to specify the conditional features.
+It will accept similar types to the `sensitive_features=` argument.
+Suppose we have another column called `income_level` with unique values 'Low' and 'High'
+```python
+>>> metric = GroupedMetric(skm.accuracy_score, y_true, y_pred,
+                           sensitive_features=A_1,
+                           conditional_features=income_level)
+>>> metric.overall
+high 0.6
+low  0.4
+dtype: float64
+>>> metric.by_group
+  accuracy_score
+            high   low
+B           0.40  0.50
+C           0.55  0.65
+```
+The `overall` property is now a Pandas series, indexed by the conditional feature values.
+Similarly, the result DataFrame now uses a Pandas MultiIndex for the columns, giving one column for each (combination of) conditional feature.
+
```

Comment:
I was thinking about the following two variants for the disaggregated metrics:

![conditional_disaggregated_example](https://user-images.githubusercontent.com/20565444/92967781-dca6dd80-f447-11ea-912f-bd96ef417475.png)